### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by mrueg**
This change bumps golang to 1.16.12 and updates dependencies in go.mod.

### CODE REVIEW

**Comment by jcmoraisjr** [comment]
Great, thank you! We should have a v0.14 tag in a month or so, I'll leave this open so we can merge an up to date client-go and golang versions.

**Comment by mrueg** [comment]
> Great, thank you! We should have a v0.14 tag in a month or so, I'll leave this open so we can merge an up to date client-go and golang versions.



Do you want me to upgrade to go 1.17 and k8s 1.23?

**Comment by jcmoraisjr** [comment]
Both golang and Kubernetes releases have a faster step if compared with haproxy ingress, so yes go 1.17 and k8s 1.23 seems to be a more reasonable options.

**Comment by jcmoraisjr** [comment]
Hi, we're really late with v0.14 and maybe we'll issue the first tag in the next few days. Please confirm if you can squash all the commits into a single one and rebase on top of master, I'll merge it just after that. Btw thanks for maintaining this branch up to date.

**Comment by mrueg** [comment]
@jcmoraisjr working on fixing conflicts and squashing

File: builder/Dockerfile [hunk 1]
```diff
@@ -12,16 +12,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.16.5-alpine AS builder
+FROM golang:1.17.6-alpine AS builder
 
 ARG GIT_TAG
-
+WORKDIR /src
```
Comment (mrueg) [inline builder/Dockerfile:18]: @jcmoraisjr small additional change, it's preferred over using cd

Comment (jcmoraisjr) [reply builder/Dockerfile:18]: sure, sounds good.

**Comment by jcmoraisjr** [comment]
My local env suggested me to move `gopkg.in/go-playground/assert.v1` to the second require argument, can you confirm this?

**Comment by mrueg** [comment]
> My local env suggested me to move `gopkg.in/go-playground/assert.v1` to the second require argument, can you confirm this?

Yes, you're correct. I fixed it! Thanks for catching that :)

**Comment by jcmoraisjr** [comment]
LGTM, thanks! Merging now.

**Pull request merged**